tensorbay/opendataset/nuScenes/loader.py

```diff
@@ -0,0 +1,303 @@
+        dataset.add_segment(segment)
+
+
+def _load_info(filepath: str, key: str = "token") -> Dict[str, Any]:
+    with open(filepath, "r") as fp:
+        info = json.load(fp)
+    if key == "token":
+        info_with_keys = {item[key]: item for item in info}
+    else:
+        info_with_keys = {}
+        for item in info:
+            info_with_keys.setdefault(item[key], []).append(item)
+    return info_with_keys
+
+
+def _load_key_frame_info(filepath: str) -> Dict[str, Any]:
+    with open(filepath, "r") as fp:
+        info = json.load(fp)
+    info_with_keys: Dict[str, Any] = {}
+    for item in info:
+        if item["is_key_frame"]:
+            info_with_keys.setdefault(item["sample_token"], []).append(item)
+    return info_with_keys
+
+
+def _generate_segments(
+    dataset: FusionDataset, scene_file_path: str, phase: str
+) -> Iterator[Tuple[FusionSegment, str]]:
+    with open(scene_file_path, "r") as file:
+        scene_info = json.load(file)
+    for scene in scene_info:
+        segment = dataset.create_segment(f"{phase}/{scene['name']}")
+        segment.description = scene["description"]
+        yield segment, scene["first_sample_token"]
+
+
+def _load_frame(
+    current_frame_data: List[Dict[str, Any]],
+    calibrated_sensors: Dict[str, Any],
+    general_knowledge: Dict[str, Dict[str, Any]],
+    sensors: Sensors,
+    phase_path: str,
+) -> Tuple[Frame, str]:
+    frame = Frame()
+    lidar_ego_pose_token: str
+    for sensor_frame in current_frame_data:
+        calibrated_sensor_info = calibrated_sensors[sensor_frame["calibrated_sensor_token"]]
+        common_sensor = general_knowledge["sensor"][calibrated_sensor_info["sensor_token"]]
+        sensor_name = common_sensor["channel"]
+        sensor_type = common_sensor["modality"]
+
+        if sensor_type == "lidar":
+            lidar_ego_pose_token = sensor_frame["ego_pose_token"]
+
+        if sensor_name not in sensors:
+            sensors.add(_load_sensor(sensor_type, sensor_name, calibrated_sensor_info))
+
+        filepath = os.path.join(phase_path, sensor_frame["filename"])
+        data = Data(filepath, timestamp=sensor_frame["timestamp"] / 10 ** 6)
+
+        frame[sensor_name] = data
+    return frame, lidar_ego_pose_token
+
+
+def _load_sensor(
+    sensor_type: str, sensor_name: str, calibrated_sensor_info: Dict[str, Any]
+) -> Union[Lidar, Radar, Camera]:
+    sensor = SENSOR_TYPE_CLASS[sensor_type](sensor_name)
+    sensor.set_extrinsics(
+        translation=calibrated_sensor_info["translation"],
+        rotation=calibrated_sensor_info["rotation"],
+    )
+    intrinsics = calibrated_sensor_info["camera_intrinsic"]
+    if intrinsics:
+        sensor.set_camera_matrix(matrix=intrinsics)  # type: ignore[union-attr]
+    return sensor
+
+
+def _load_labels(
+    sample_annotation: List[Dict[str, Any]],
+    instance_info: Dict[str, Any],
+    sensor_to_ego: Transform3D,
+    lidar_ego_pose_info: Dict[str, Any],
```

Comment:
convert `lidar_ego_pose_info` to `Transform3D` before passing it into this function?